### PR TITLE
Ecdc 1705 save histogram images

### PR DESCRIPTION
**Pull request by kmurica**
Ready for review. Notice that the only format that seems completely good visually is .pdf on my mac.
Linux also has issues with other formats (png, jpeg) where the axes disappear in the saved file.
As far as I see, this seems to be related to the gr python package, but I haven't been able to successfully find a workaround.
Some help from MLZ will be needed to debug the issues.

### CODE REVIEW

**Comment by umithardal** [comment]
@kmurica could you please add reviewers (top right)? Otherwise we do not get notifications (at least I don't). Thanks :)

**Comment by kmurica** [comment]
> @kmurica could you please add reviewers (top right)? Otherwise we do not get notifications (at least I don't). Thanks :)

Done! :)

File: nicos/guisupport/livewidget/__init__.py [hunk 1]
```diff
@@ -366,6 +368,41 @@ def setCenterMark(self, flag):
         self.axes.drawxylines = flag
         self.gr.update()
 
+    def savePlot(self):
```
Comment (umithardal) [inline nicos/guisupport/livewidget/__init__.py:371]: Let us not use Camel naming, instead use `_` in between the words with all small letters, in function as well as parameter names. This is mainly for readability.

File: nicos/guisupport/livewidget/__init__.py [hunk 2]
```diff
@@ -366,6 +368,41 @@ def setCenterMark(self, flag):
...
+                            break
+                if suffix and (suffix[1:] in gr.PRINT_TYPE or
+                               suffix[1:] in gr.GRAPHIC_TYPE):
+                    self.gr.save(path)
+                    self._saveName = os.path.basename(path)
+                else:
+                    self._saveName = None
+                    raise Exception("Unsupported file format")
+        return self._saveName
+
```
Comment (umithardal) [inline nicos/guisupport/livewidget/__init__.py:405]: It is very hard to follow what is happening in this function. Could you please add some comments and may be a doc-string?

File: nicos/guisupport/livewidget/__init__.py [hunk 3]
```diff
@@ -366,6 +368,41 @@ def setCenterMark(self, flag):
...
+        for prtype in [gr.PRINT_JPEG, gr.PRINT_TIF]:
+            dictPrintType.pop(prtype)
+        saveTypes = (";;".join(dictPrintType.values()) + ";;" +
+                     ";;".join(gr.GRAPHIC_TYPE.values()))
+
+        dialog = QFileDialog(self, "Select file name", "", saveTypes)
+        dialog.selectNameFilter(gr.PRINT_TYPE[gr.PRINT_PDF])
+        dialog.setOption(dialog.HideNameFilterDetails, False)
+        dialog.setAcceptMode(QFileDialog.AcceptSave)
+        if dialog.exec_() == QDialog.Accepted:
```
Comment (umithardal) [inline nicos/guisupport/livewidget/__init__.py:383]: We should find a Pythonic way to write the code. These `if` ladders are bug prone and hard to maintain. Also, may be not so performance efficient?

**Comment by mattclarke** [comment]
Can you put this Gerrit please?
Then we can discuss the issues with the axes with the FRM guys.
Put `[WIP]` as the front of the commit message, so they know it is a work in progress.

File: nicos/guisupport/livewidget/__init__.py (hunk 3 above)
Comment (kmurica) [reply nicos/guisupport/livewidget/__init__.py]: I have now simplified the code a lot. It should be easier to follow it now.

File: nicos/guisupport/livewidget/__init__.py (hunk 2 above)
Comment (kmurica) [reply nicos/guisupport/livewidget/__init__.py:386]: Even if the function is simplified now, I have added documentation to the function.

File: nicos/guisupport/livewidget/__init__.py (hunk 1 above)
Comment (kmurica) [reply nicos/guisupport/livewidget/__init__.py:370]: I agree and I have done it for the function variables. But in order to keep it consistent with the class implementation as a whole, I have kept the "savePlot" naming. Any major objection to this? My reasoning is to avoid mixing camel naming and _ naming in the same class.

**Comment by kmurica** [comment]
This task is approved and merged in Gerrit. Closing this PR.
